ftw/linkchecker/command/checking_links.py

```diff
@@ -0,0 +1,94 @@
+from AccessControl.SecurityManagement import newSecurityManager
+from zc.relation.interfaces import ICatalog
+from zope.component import getUtility
+from zope.component.hooks import setSite
+import AccessControl
+
+
+def get_plone_sites_information(app):
+
+    """Get all PloneSites and their information
+    """
+    def get_plone_sites(app):
+        for child in app.objectValues():
+            if child.meta_type == 'Plone Site':
+                yield child
+            elif child.meta_type == 'Folder':
+                for item in get_plone_sites(child):
+                    yield item
+
+    plone_sites = get_plone_sites(app)
+
+    plone_sites_information = [{
```

Comment:
I do not see a good enough reason to iterate over every PloneSite and only return this 3 information. For another developer who sees an element of this list in another function eg: `get_broken_relations_in_plone_page(app, site_info)` he will not know what keys are available. If you'd just return a list of `PloneSite`'s (or generator as in `get_plone_sites`) and would call the `get_broken_relations_in_plone_page` function with `plone_site` instead of `site_info` everyone would know, that it is a `PloneSite` and has all the available information.

Also if you iterate over the generator `get_plone_sites` as in `main`, you'd speed up the initial the initial call, because you do not iterate over every object but only search for the next object after you have completed your operations on the first one.

Reply:
I'll take care of that in #18, so we can proceed with this pull request.